core/jvm/src/main/scala/zio/sql/expr.scala

```diff
@@ -158,6 +158,10 @@ trait ExprModule extends NewtypesModule with FeaturesModule with OpsModule {
       def typeTag: TypeTag[Z] = implicitly[TypeTag[Z]]
     }
 
+    sealed case class FunctionCall0[F, A, B, Z: TypeTag](function: FunctionDef[B, Z]) extends InvariantExpr[F, A, Z] {
```

Comment:
I think this should be 
```scala
sealed case class FunctionCall0[Z: TypeTag](function: FunctionDef[Any, Z]) extends InvariantExpr[Features.Function0, Any, Z]
```
since we don't care what the input type is (there is no input for a Function0) and we're introducing a new Feature that isn't based on anything else (see other comment re Features.Function0)

Reply:
type changes? It shouldn't be possible to create a `FunctionCall0` that has features other than `Features.FunctionCall0` and an input type other than `Any`, and the `FunctionDef` needs to be our equivalent of `Any => Z`